plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java

```diff
@@ -70,11 +72,47 @@ public class VFSFileProvider extends BaseFileProvider<VFSFile> {
 
   public static final String NAME = "VFS Connections";
   public static final String TYPE = "vfs";
-  public static final String DOMAIN_ROOT = "[\\w]+://";
 
-  private Bowl bowl = DefaultBowl.getInstance();
+  /**
+   * VFS Connection scheme or prefix
+   */
+  public static final String CONNECTION_SCHEME = ConnectionFileSystem.DOMAIN_ROOT;
+
+  /**
+   * Regular expression for connection name
+   */
+  protected static final String CONNECTION_NAME_REGEX = "[\\w_-]+";
```

Comment:
You can create a VFS connection with special characters such as "###" why, Idk. But I should be in sync with the other parts of the Kettle VFS UI. Let me see if I can find the kettle code that puts any restrictions on the connection name, otherwise, I'll just allow anything on my part and consider it the duty of other parts of the code to manage the allowed characters.

Reply:
I agree there does need to be a change, but the real question is what is the destination. I started an offline slack thread  https://hitachivantara-eng.slack.com/archives/C01HJ874L8Z/p1715615047059579

Reply:
Addressed see commit https://github.com/pentaho/pentaho-kettle/commit/3e8bcc9f08d5634a303f5bdcbaf73f80d1811e46

I created the class `ConnectionUriParser.java` that handles this issue. Also, see associated unit test class.

As discussed offline, the manager will tell us the exact set of special characters to handle.